### PR TITLE
Complete Phase 2 of Deprecation for MetagenomeSequencing class

### PR DESCRIPTION
**Pull request by hesspnnl**
Complete phase 2 of depreacation as the [deprecation guide](https://github.com/microbiomedata/nmdc-schema/blob/34717ed8042412b96ab1db132037941258d9a70a/src/docs/schema_element_deprecation_guide.md) outlines

Actions:
- Move the MetagenomeSequencing class to the "deprecated.yaml" schema file
- Remove MetagenomeSequencing class from the original schema YAML file (e.g. core.yaml, annotation.yaml, etc.).
- Remove any references to the deprecated element from the original schema YAML file. 
- Delete invalid and valid MetagenomeSequencing yaml files

### CODE REVIEW

**Comment by hesspnnl** [comment]
@kheal I am not sure how to remove the reference here in these two spots:

- https://github.com/microbiomedata/nmdc-schema/blob/1261-complete-phase-2-of-deprecation-for-nmdcmetagenomesequencing/src/data/valid/Database-library-prep-exhausive.yaml#L1 
- https://github.com/microbiomedata/nmdc-schema/blob/1261-complete-phase-2-of-deprecation-for-nmdcmetagenomesequencing/src/data/valid/Database-neon-story.yaml#L42

File: nmdc_schema/migrators/partials/migrator_from_10_2_0_to_11_0_0/migrator_from_10_2_0_to_11_0_0_part_10.py [hunk 1]
```diff
@@ -118,7 +117,6 @@ def upgrade(self):
             pooling_set=[lambda document: self.add_type_slot_with_class_uri(document, "nmdc:Pooling", slots_with_inlined_classes)],
             metagenome_annotation_set=[lambda document: self.add_type_slot_with_class_uri(document, "nmdc:MetagenomeAnnotation", slots_with_inlined_classes)],
             metagenome_assembly_set=[lambda document: self.add_type_slot_with_class_uri(document, "nmdc:MetagenomeAssembly", slots_with_inlined_classes)],
-            metagenome_sequencing_set=[lambda document: self.add_type_slot_with_class_uri(document, "nmdc:MetagenomeSequencing", slots_with_inlined_classes)],
```
Comment (kheal) [inline nmdc_schema/migrators/partials/migrator_from_10_2_0_to_11_0_0/migrator_from_10_2_0_to_11_0_0_part_10.py:121]: @hesspnnl I don't think we change migrators retrospectively.  Check with Eric.

**Comment by kheal** [comment]
> @kheal I am not sure how to remove the reference here in these two spots:
> 
> * https://github.com/microbiomedata/nmdc-schema/blob/1261-complete-phase-2-of-deprecation-for-nmdcmetagenomesequencing/src/data/valid/Database-library-prep-exhausive.yaml#L1
> * https://github.com/microbiomedata/nmdc-schema/blob/1261-complete-phase-2-of-deprecation-for-nmdcmetagenomesequencing/src/data/valid/Database-neon-story.yaml#L42

^ I don't see any reference to the deprecated class in either of these, so I don't think any changes are necessary.

File: nmdc_schema/migrators/partials/migrator_from_10_2_0_to_11_0_0/migrator_from_10_2_0_to_11_0_0_part_10.py [hunk 2]
```diff
@@ -118,8 +117,8 @@ def upgrade(self):
...
             nom_analysis_set=[lambda document: self.add_type_slot_with_class_uri(document, "nmdc:NomAnalysis", slots_with_inlined_classes)],
+            metagenome_sequencing_set=[lambda document: self.add_type_slot_with_class_uri(document, "nmdc:MetagenomeSequencing", slots_with_inlined_classes)],
```
Comment (eecavanna) [inline nmdc_schema/migrators/partials/migrator_from_10_2_0_to_11_0_0/migrator_from_10_2_0_to_11_0_0_part_10.py:121]: Hi @hesspnnl, I'd prefer this line be restored into its original location. Also, line 81 was modified (the line that says `"MetagenomeSequencing"` was removed). I want that line to be restored also. That way, the overall diff for the PR doesn't show any changes to this file.

**Comment by kheal** [comment]
@hesspnnl and @eecavanna  - If I'm reading the failing tests correctly it is related to the doctest of the old migrator (`10_2_0_to_11_0_0`) that Olivia reverted changes for.  In order to get these tests to pass, we will need to adjust or remove the doctest of that migrator (though we can keep the actual function of the migrator in the repo as is.  I consider this is a weird edge case and not the norm.

File: nmdc_schema/migrators/partials/migrator_from_10_2_0_to_11_0_0/migrator_from_10_2_0_to_11_0_0_part_10.py [hunk 3]
```diff
@@ -27,39 +27,48 @@ def upgrade(self):
...
-        >>> library_prep_4['protocol_link']['type']
-        'nmdc:Protocol'
+        Note: We added a prefix, `SKIP: `, to the doctests in this method to prevent them from being run.
+              Reason for skipping: The tests target a function that—through its use of a `SchemaView` that
+              is bound to the schema currently in the repository—requires that specific schema elements
+              be defined in the schema, and those elements are not defined in the schema anymore. Note that
+              the prefix string is arbitrary (I chose `SKIP: ` in an attempt to make it self-documenting).
+
+        >>> from sys import stderr
+        >>> print("⚠️  Skipping doctests", file=stderr)
```
Comment (eecavanna) [inline nmdc_schema/migrators/partials/migrator_from_10_2_0_to_11_0_0/migrator_from_10_2_0_to_11_0_0_part_10.py:37]: These lines make it so the typical doctest output includes the following line:

image

**Review by eecavanna — APPROVED** [review]
I approve of the migrator changes (😅).

**Review by kheal — APPROVED** [review]

**Review by turbomam — APPROVED** [review]
I approve and would like to include this commit:
- https://github.com/microbiomedata/nmdc-schema/pull/2579/commits/e16c1fc89d8679886750e71fdeafc555f1151676
